Software/api/src/main/java/org/foi/air/api/request_handlers/RegistrationRequestHandler.kt

```diff
@@ -0,0 +1,68 @@
+package org.foi.air.api.request_handlers
+
+import ResponseListener
+import com.google.gson.Gson
+import okhttp3.ResponseBody
+import org.foi.air.api.models.RegistrationBody
+import org.foi.air.api.network.ApiService
+import org.foi.air.core.data_classes.UserInfo
+import org.foi.air.core.network.RequestHandler
+import org.foi.air.core.models.ErrorResponseBody
+import org.foi.air.core.models.SuccessfulRegisterResponseBody
+import org.json.JSONException
+import org.json.JSONObject
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class RegistrationRequestHandler(private val requestBody: RegistrationBody):
+    RequestHandler<SuccessfulRegisterResponseBody> {
+     override fun sendRequest(responseListener: ResponseListener<SuccessfulRegisterResponseBody>) {
+        val service = ApiService.authService
+        val serviceCall = service.registerUser(requestBody)
+
+        serviceCall.enqueue(object : Callback<ResponseBody>{
+
+            override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
+                if (response.isSuccessful) {
+                    if (response.body() != null) {
+                        val convertedResponse = successfulResponseConverter(response.body()!!)
+                        if(convertedResponse == null){
+                            responseListener.onErrorResponse(ErrorResponseBody(false,"Registered successfully, but failed to convert request data. Please go to Login page and login.","Unsuccessful request conversion"))
```

Comment:
I don't think it's a good idea to include "failed to convert request data" in the error. Most users don't know what that means and they have no use for that information. It would be better to write something along the lines of "Registered successfully, please go to login page".